src/ethereum/osaka/fork.py

```diff
@@ -0,0 +1,1003 @@
+    transaction_fee = tx_gas_used * priority_fee_per_gas
+
+    # refund gas
+    sender_balance_after_refund = get_account(
+        block_env.state, sender
+    ).balance + U256(gas_refund_amount)
+    set_account_balance(block_env.state, sender, sender_balance_after_refund)
+
+    # transfer miner fees
+    coinbase_balance_after_mining_fee = get_account(
+        block_env.state, block_env.coinbase
+    ).balance + U256(transaction_fee)
+    if coinbase_balance_after_mining_fee != 0:
+        set_account_balance(
+            block_env.state,
+            block_env.coinbase,
+            coinbase_balance_after_mining_fee,
+        )
+    elif account_exists_and_is_empty(block_env.state, block_env.coinbase):
+        destroy_account(block_env.state, block_env.coinbase)
+
+    for address in tx_output.accounts_to_delete:
+        destroy_account(block_env.state, address)
+
+    destroy_touched_empty_accounts(block_env.state, tx_output.touched_accounts)
+
+    block_output.block_gas_used += tx_gas_used
+    block_output.blob_gas_used += tx_blob_gas_used
+
+    receipt = make_receipt(
+        tx, tx_output.error, block_output.block_gas_used, tx_output.logs
+    )
+
+    trie_set(
+        block_output.receipts_trie,
+        rlp.encode(Uint(index)),
+        receipt,
+    )
+
+    block_output.block_logs += tx_output.logs
+
+    block_output.deposit_requests += parse_deposit_requests_from_receipt(
+        receipt
+    )
+
+
+def process_withdrawals(
+    block_env: vm.BlockEnvironment,
+    block_output: vm.BlockOutput,
+    withdrawals: Tuple[Withdrawal, ...],
+) -> None:
+    """
+    Increase the balance of the withdrawing account.
+    """
+
+    def increase_recipient_balance(recipient: Account) -> None:
+        recipient.balance += wd.amount * U256(10**9)
+
+    for i, wd in enumerate(withdrawals):
+        trie_set(
+            block_output.withdrawals_trie,
+            rlp.encode(Uint(i)),
+            rlp.encode(wd),
+        )
+
+        modify_state(block_env.state, wd.address, increase_recipient_balance)
+
+        if account_exists_and_is_empty(block_env.state, wd.address):
+            destroy_account(block_env.state, wd.address)
+
+
+def validate_inclusion_list(
+    block_env: vm.BlockEnvironment,
+    block_output: vm.BlockOutput,
+    transactions: Tuple[Union[Bytes, LegacyTransaction], ...],
+    inclusion_list: Tuple[Union[Bytes, LegacyTransaction], ...],
+) -> None:
+    """
+    Validate the block satisfies the inclusion list.
+    """
+
+    for tx in inclusion_list:
+        # If the transaction is already present in the block, then skip.
+        if tx in transactions:
+            continue
+
+        block_env = deepcopy(block_env)
+        block_output = deepcopy(block_output)
+        try:
+            tx = decode_transaction(tx)
+            index = Uint(len(transactions))
+            process_transaction(block_env, block_output, tx, index)
```

Comment:
We don't execute IL transactions. We just need to check its nonce and balance, and see if there is sufficient gas left to include any missing IL transactions.

Current version of EIP is misleading and we have a pending PR that fixes it: https://github.com/ethereum/EIPs/pull/9381/files

Reply:
ah I see. I forgot about this PR, and so I was just going off what I saw in the EIP.

Reply:
I would say in addition to checking for nonce and balance, would we need to perform some basic checks to make sure that the transaction is well formed? Including perhaps the checks that are performed in the [validate_transaction](https://github.com/ethereum/execution-specs/blob/68f3a1f1ccebc048daf5dfec58bf0325e5fcc1b8/src/ethereum/osaka/transactions.py#L191) function?

Reply:
The EL builds an IL with transactions from the mempool and each transaction is validated when added to the mempool. That being said, it makes sense to specify that IL transactions are intrinsically valid.